aten/src/ATen/native/nested/NestedTensorMath.cpp

```diff
@@ -60,15 +60,23 @@ Tensor pad_tensor_to_shape(
 
 
 Tensor NestedTensor_nested_tensor_from_mask(const Tensor& t, const Tensor& mask, bool mask_check) {
-    TORCH_CHECK(mask.scalar_type() == at::ScalarType::Bool, "Expected mask to be of ScalarType Bool, but got ", mask.scalar_type(), " instead.");
+  TORCH_CHECK((mask.scalar_type() == at::ScalarType::Bool) || (at::is_floating_point(mask)),
+                "Expected mask to be of type Bool or floating dtype but got ",
+                mask.scalar_type(), " instead.");
     TORCH_CHECK(mask.dim() == 2, "Padding mask should be 2D");
     TORCH_CHECK(t.dim() == 3, "Input should be a 3D tensor, N * L * D");
     auto N = t.size(0), L = t.size(1), D = t.size(2);
     auto NN = mask.size(0), LL = mask.size(1);
     TORCH_CHECK(N == NN && L == LL, "Mask size should match input size");
 
     // N * L
-    Tensor sizes = mask;
+    Tensor sizes;
+    if (mask.scalar_type() != at::ScalarType::Bool){
```

Comment:
hm, why do we still need to allow floats here and in `NestedTensor_nested_tensor_from_mask_left_aligned` if we do `src_key_padding_mask_.logical_not()` when passing the mask argument in which iiuc will return a bool mask below?

Reply:
Removed.